### ISSUE TITLE
Make use of Memoizer to cache metadata for subsequent reads

### ISSUE THREAD

**Issue by imagejan** [scifio/scifio-bf-compat#20]
Any SCIFIO format wrapping around Bio-Formats should ideally make use of [`loci.formats.Memoizer`](https://javadoc.scijava.org/Bio-Formats/index.html?loci/formats/Memoizer.html) to speed up subsequent reads of the same file/dataset.

Having a default reader plugin with this functionality for all file formats supported by Bio-Formats would greatly benefit the community. See also [this discussion on the image.sc forum](https://forum.image.sc/t/speeding-up-metamorph-file-reading-with-bioformats/42532?u=imagejan).

@hinerm, @ctrueden would you have pointers how to best approach this?

**Comment by ctrueden** [comment]
I agree that it makes sense to tackle this as part of #12. I do not have insight into how to begin, without digging into the code myself, which I will not have time to do before next year, sorry.

**Comment by imagejan** [comment]
As #12 (splitting into separate readers/writers per file format) seems more involved, an immediate option might be to change this line:

https://github.com/scifio/scifio-bf-compat/blob/59303c255e662ed46a61b33bd307f7894360491e/src/main/java/io/scif/bf/BioFormatsFormat.java#L461

... to something like:

```java
meta.setReader(new Memoizer(reader));
```

... and see whether it works. 🤷  🙂 

Will try when I find the time.
